specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json

```diff
@@ -2593,7 +2901,330 @@
       ],
       "description": "List of Vpn-Sites."
     },
-    "VirtualHubProperties": {
+    "VpnServerConfigVpnClientRootCertificatePropertiesFormat": {
+      "properties": {
+        "publicCertData": {
+          "type": "string",
+          "description": "The certificate public data."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The provisioning state of the VpnServerConfiguration VPN client root certificate resource. Possible values are: 'Updating', 'Deleting', and 'Failed'."
+        }
+      },
+      "required": [
+        "publicCertData"
+      ],
+      "description": "Properties of VPN client root certificate of VpnServerConfiguration."
+    },
+    "VpnServerConfigVpnClientRootCertificate": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/VpnServerConfigVpnClientRootCertificatePropertiesFormat",
+          "description": "Properties of the VpnServerConfiguration VPN client root certificate."
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the resource that is unique within a resource group. This name can be used to access the resource."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "required": [
+        "properties"
+      ],
+      "description": "VPN client root certificate of VpnServerConfiguration."
+    },
+    "VpnServerConfigRadiusServerRootCertificatePropertiesFormat": {
+      "properties": {
+        "publicCertData": {
+          "type": "string",
+          "description": "The certificate public data."
+        },
+        "provisioningState": {
```

Comment:
What is the function of `provisioningState` on the certificates? Is that used to indicate status of some kind of async instantiation? Does the `provisioningState` property of vpnServerConfigurations resource reflect these in aggregate?

Reply:
Provisioning state on the certificate shows the state of that certificate child resource. This is child resource of top level resource: vpnserverconfiguration so they have provisioning state as well in NRP.